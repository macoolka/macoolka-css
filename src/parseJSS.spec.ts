
import {  parse} from '.';

describe('parseJSS', () => {
    it('parse mkHide', () => {
        expect(parse({
            mkHide: ['small'],

        })).toMatchSnapshot();
    });
    it('parse stand', () => {
        expect(parse({
            mkPadding: 'small',
            height: 100,
            selector: {
                '&:focus': {
                    mkPadding: 'medium',
                    height: 200,
                }
            }
        })).toMatchSnapshot();
    });
    it('parse media', () => {
        expect(parse({
            height: 100,
            mkMedia: [{
                mkPadding: 'small',
                fontSize: 12
            }, {
                mkPadding: 'medium', fontSize: 13
            }, {
                mkPadding: 'large', fontSize: 14
            }, {
                mkPaddingTop: 'large', fontSize: 15
            }],
            selector: {
                '&:focus': {
                        mkColor: 'main',
                        borderWidth: 1
                }
            }
        })).toMatchSnapshot();
    });

});
describe('parse', () => {
    it('parse stand', () => {
        expect(parse({
            mkPadding: 'small',
            height: 100,
            selector: {
                '&:focus': {
                    mkPadding: 'medium',
                    height: 200,
                }
            }
        })).toMatchSnapshot();
    });
    it('parse media', () => {
        expect(parse({
            height: 100,
            mkMedia: [{
                mkPadding: 'small',
                fontSize: 12
            }, {
                mkPadding: 'medium', fontSize: 13
            }, {
                mkPadding: 'large', fontSize: 14
            }, {
                mkPadding: 'medium', fontSize: 15
            }],
            selector: {
                '&:focus': {
                    mkColor: 'main',
                    borderWidth: 1
            }
            }
        })).toMatchSnapshot();
    });

});