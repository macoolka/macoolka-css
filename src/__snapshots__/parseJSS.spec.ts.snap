// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse parse media 1`] = `
"height: 100px;
&:focus {
  border-width: 1px;
  background-color: #009688;
  color: #fff;
}
@media screen and (max-width: 120em) {
  font-size: 15px;
  padding: 16px;
}
@media screen and (max-width: 93em) {
  font-size: 14px;
  padding: 24px;
}
@media screen and (max-width: 80em) {
  font-size: 13px;
  padding: 16px;
}
@media screen and (max-width: 50em) {
  font-size: 12px;
  padding: 8px;
}
"
`;

exports[`parse parse stand 1`] = `
"padding: 8px;
height: 100px;
&:focus {
  padding: 16px;
  height: 200px;
}
"
`;

exports[`parseJSS parse media 1`] = `
"height: 100px;
&:focus {
  border-width: 1px;
  background-color: #009688;
  color: #fff;
}
@media screen and (max-width: 120em) {
  font-size: 15px;
  padding-top: 24px;
}
@media screen and (max-width: 93em) {
  font-size: 14px;
  padding: 24px;
}
@media screen and (max-width: 80em) {
  font-size: 13px;
  padding: 16px;
}
@media screen and (max-width: 50em) {
  font-size: 12px;
  padding: 8px;
}
"
`;

exports[`parseJSS parse mkHide 1`] = `
"mk-hide: small;

"
`;

exports[`parseJSS parse stand 1`] = `
"padding: 8px;
height: 100px;
&:focus {
  padding: 16px;
  height: 200px;
}
"
`;
